source/client-side-encryption/client-side-encryption.rst

```diff
@@ -498,13 +544,68 @@ detail described in the `Implementation`_ section:
 
       // Passed when spawning mongocryptd. If omitted, this defaults to ["--idleShutdownTimeoutSecs=60"]
       mongocryptdSpawnArgs: Optional<Array[String]>
+
+      // Override the path used to load the csfle library
+      csflePath: Optional<string>;
+
+      // If 'true', refuse to continue encryption without a csfle library
+      csfleRequired: boolean;
+
+      // If 'true', libmongocrypt will not search for csfle
+      cslfeDisabled: boolean;
    }
 
 Drivers MUST implement extraOptions in a way that allows
 deprecating/removing options in the future without an API break, such as
 with a BSON document or map type instead of a struct type with fixed
 fields. See `Why are extraOptions and kmsProviders maps?`_.
 
+.. option:: extraOptions.csflePath
+
+   :type: :ts:`undefined | string`
+   :default: :ts:`undefined`
+
+   .. |opt-path-override| replace:: :option:`extraOptions.csflePath`
+
+   Allow the user to specify an absolute path to a csfle_ dynamic library to
+   load. Refer:
+
+   - `Overriding the csfle Library Path`_
+   - `Path Resolution Behavior`_
+   - `Enabling csfle`_
+
+
+.. option:: extraOptions.csfleRequired
+
+   :type: :ts:`boolean`
+   :default: |false|
+
+   .. |opt-csfle-required| replace:: :option:`extraOptions.csfleRequired`
+
+   If |true|, the driver MUST refuse to continue unless csfle_ was loaded
+   successfully.
+
+   If, after initializing a `libmongocrypt_handle`, csfle_ is detected to be
+   unavailable AND |opt-csfle-required| is |true|, the driver
+   MUST consider the `libmongocrypt_handle` to be invalid and return an error to
+   the user. Refer:
+
+   - `Enabling csfle`_
+   - `Managing mongocryptd`_
+   - `Detecting csfle Availability`_
+
+
+.. option:: extraOptions.csfleDisabled
```

Comment:
Please consider removing this option.

Rationale: in the near future, I would like to be able to remove support for mongocryptd entirely (in both server and drivers).  An API like this (even if only in extraOptions) could make that more difficult, as it may require a round of deprecations/major releases to get there.  If we leave it out, the only new options are ones that could more easily be claimed to be supported in a future worlds with no mongocryptd (or even a world where clfle is compiled into libmongocrypt).

We can always add this option in the future if its requested by users experiencing actual deployment pain.